fluid/neural_machine_translation/rnn_search/train.py

```diff
@@ -0,0 +1,149 @@
+"""seq2seq model for fluid."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+import argparse
+import time
+
+import paddle.v2 as paddle
+import paddle.fluid as fluid
+import paddle.fluid.core as core
+import paddle.fluid.framework as framework
+from paddle.fluid.executor import Executor
+
+from model import seq_to_seq_net
+from config import TrainConfig as train_conf
+
+
+def to_lodtensor(data, place):
+    seq_lens = [len(seq) for seq in data]
+    cur_len = 0
+    lod = [cur_len]
+    for l in seq_lens:
+        cur_len += l
+        lod.append(cur_len)
+    flattened_data = np.concatenate(data, axis=0).astype("int64")
+    flattened_data = flattened_data.reshape([len(flattened_data), 1])
+    lod_t = core.LoDTensor()
+    lod_t.set(flattened_data, place)
+    lod_t.set_lod([lod])
+    return lod_t, lod[-1]
+
+
+def lodtensor_to_ndarray(lod_tensor):
+    dims = lod_tensor.get_dims()
+    ndarray = np.zeros(shape=dims).astype('float32')
+    for i in xrange(np.product(dims)):
+        ndarray.ravel()[i] = lod_tensor.get_float_element(i)
+    return ndarray
+
+
+def train():
+    src_word_idx = fluid.layers.data(
+        name='source_sequence', shape=[1], dtype='int64', lod_level=1)
+    trg_word_idx = fluid.layers.data(
+        name='target_sequence', shape=[1], dtype='int64', lod_level=1)
+    label = fluid.layers.data(
+        name='label_sequence', shape=[1], dtype='int64', lod_level=1)
+    if train_conf.parallel:
+        places = fluid.layers.get_places()
+        pd = fluid.layers.ParallelDo(places)
+        with pd.do():
+            src_word_idx_ = pd.read_input(src_word_idx)
+            trg_word_idx_ = pd.read_input(trg_word_idx)
+            label_ = pd.read_input(label)
+            avg_cost = seq_to_seq_net(src_word_idx_, trg_word_idx_, label_)
+            pd.write_output(avg_cost)
+        avg_cost = pd()
+        avg_cost = fluid.layers.mean(x=avg_cost)
+    else:
+        avg_cost = seq_to_seq_net(src_word_idx, trg_word_idx, label)
+
+    feeding_list = ["source_sequence", "target_sequence", "label_sequence"]
+
+    inference_program = fluid.default_main_program().clone()
+
+    optimizer = fluid.optimizer.Adam(learning_rate=train_conf.learning_rate)
+    optimizer.minimize(avg_cost)
+
+    train_batch_generator = paddle.batch(
+        paddle.reader.shuffle(
+            paddle.dataset.wmt16.train(train_conf.source_dict_dim,
+                                       train_conf.target_dict_dim),
+            buf_size=1000),
```

Comment:
please make `buf_size` larger, for example, 100000.

Reply:
Done